lib/Service/ApiService.php

```diff
@@ -67,4 +81,35 @@ public function sendFile(int $fileId): void {
 			],
 		);
 	}
+
+	public function searchMessages(string $userId, string $term, int $offset = 0, int $limit = 10): array {
```

Comment:
You are still not implementing the pagination that is necessary for properly searching.

Reply:
Actually, I don't see any pagination feature in the API documentation. So, instead of implementing at the API call, the pagination feature is implemented manually here in the `SearchProvider.php`.

``` 
// Paginate the results manually since the API does not provide offset/limit
$pagedResults = array_slice($searchResult['results'], $offset, $limit);

$formattedResults = array_map(function (array $entry) use ($url): SearchResultEntry {
        $finalThumbnailUrl = '';
        $title = $entry['title'] ?? 'Untitled';
        $context = strip_tags($entry['__search_hit__']['highlights'] ?? '');
        $link = $this->getLinkToPaperless($entry, $url);
        return new SearchResultEntry(
                $finalThumbnailUrl,
                $title,
                $context,
                $link,
                $finalThumbnailUrl,
                true
        );
}, $pagedResults);
```

Reply:
![image](https://github.com/user-attachments/assets/c77046de-17ba-44f6-ae04-3029ed4042d1)
This is pagination.

Reply:
Hi, may I know that are you using this dev server https://github.com/juliusknorr/nextcloud-docker-dev for developing? Because currently I encountered a problem that the getLimit() function stuck at maximum value of 25, which happen to be the first page of the Paperless-ngx. I don't find any related docs online. This happens also when I am using other integration in the dev server, the result of other integration seems to be stuck at 25 too. It seems like the dev server problem, or am I missing some configuration?

Reply:
Never mind, I found the way to manually exceed the limit, guess it is hardcoded in the application.